### DISCUSSION TITLE
Support for the EEBus protocol?

### DISCUSSION THREAD

**Discussion by mtbsteve** [emsesp/EMS-ESP32#2123] [Ideas]
Would it be possible to have support for the EEBus protocol for the heatpumps in a future release?
Background is that I would like to benefit from the upcoming variable energy grid price according to the new accounting schemes Modul 1/2/3 in Germany.

**Comment by proddy** [comment]
Haven't heard of that protocol. Can you share more details?

**Reply by mtbsteve** [reply]
it’s a new standard to interconnect Smartmeters, Heatpumps, Wallboxes, Smarthome systems, Solar Energy inverters, battery storage management systems, energy providers and so on. 
https://www.eebus.org/

**Comment by weitzj** [comment]
Hi, 

I would also be interested if this is possible to implement.
UseCase: I got a SolarEdge ONE Controller. The SG-Ready standard improved from simply having a  controllable relay to use the EEBUs protocol (which is apparently already in use in EV-Chargers).

See for example the Bosch EEBus gateway to have the Bosch Heatpump EEBus compatible so that Solaredge ONE could pick it up. https://shkhelden.de/bosch-connect-key-k-40-rf-internet-gateway-und-funkmodul/7738114013

https://github.com/evcc-io/evcc/blob/master/charger/eebus.go


I understand that with esp-ems I can hook everything up via MQTT ... to HomeAssistant. But it would also be nice to be discoverable by SolarEdge ONE given the EEBus protocol is implemented.

**Reply by proddy** [reply]
We looked into the EEBus spec a while back. Since it's a very mature protocol supported by many brands. Having EMS-ESP in the middle doing the interoperability and controlling the PV systems directly makes it difficult and puts EMS-ESP into a position where we can't support or guarantee it'll work flawlessly and not mess up the heatpumps, for example. It would be better to have something like Home Assistant handle the control and communicate with EMS-ESP to send commands to the boilers.

**Reply by mtbsteve** [reply]
@proddy thanks for getting back. 
HA can already talk EEbus with some additional plugins. 
Problem is that many heatpumps do not support the EEbus protocol, especially the ones 2 or more years old. Buderus and Bosch for example haste implemented EEbus on top of the EMS bus with the new MX400 module, which is unfortunately only working with the latest generation of their heatpumps. Likely for marketing reasons they decided to not support the existing older models. 

EEbus is being used in future by the grid operators to dim the energy consumption of a heatpump to max 4.2kw in case of a local network overload. The alternative is to switch off the heatpump via the SGready interface which is a obviously a sub optimal solution. 

 Benefit for the customer is a deduction of the energy cost at least in Germany according to the §14a guidelines, which can be up to 9ct/kwh

**Comment by weitzj** [comment]
An Update from my side.  Bosch with the K40 RF key natively Supports EEBUS and it can Discover and pair to the solar edge ONE controller. Now I still need Access to the SolarEdge GO installer App to pair with the Bosch heatpump (other direction)Not succeeded, yet
